src/t_stream.c

```diff
@@ -2331,16 +2470,32 @@ void xgroupCommand(client *c) {
     streamCG *cg = NULL;
     char *opt = c->argv[1]->ptr; /* Subcommand name. */
     int mkstream = 0;
+    long long offset = 0;
     robj *o;
 
     /* CREATE has an MKSTREAM option that creates the stream if it
-     * does not exist. */
-    if (c->argc == 6 && !strcasecmp(opt,"CREATE")) {
-        if (strcasecmp(c->argv[5]->ptr,"MKSTREAM")) {
-            addReplySubcommandSyntaxError(c);
-            return;
+     * does not exist, so we want check that in advance. In this opportunity
+     * we can also parse CREATE's additional options. */
+    if (c->argc >= 6 && c->argc <= 8 && !strcasecmp(opt,"CREATE")) {
+        int i = 5;
+        while (i < c->argc) {
+            if (!strcasecmp(c->argv[i]->ptr,"MKSTREAM")) {
+                mkstream = 1;
+                i++;
+            } else if (!strcasecmp(c->argv[i]->ptr,"OFFSET") &&
+                       ++i < c->argc ) {
```

Comment:
i don't think you want the `++` to be part of the condition, i'd rather have +1 here and put the increment inside.
in the current form (since OFFSET is the last one in the if-else chain) it doesn't cause any harm, but if more options will be added, it can mess up the parsing).

nitpick: i vote against line wrapping.

Reply:
Understood and agreed.

(also unwrapped line(s)).